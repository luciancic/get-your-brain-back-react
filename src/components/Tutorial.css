.tutorial {
    display: flex;
    justify-content:space-around;
}

.tutorial-info {
    flex-basis: 35%;
}

.court{
    flex-basis: 60%;
}