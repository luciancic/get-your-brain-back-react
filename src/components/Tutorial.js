import React from 'react'

function Tutorial(props) {
    return <div className="tutorial">
        <h1>Tutorial</h1>
        <p>This video will show you how to play <em>Get Your Brain Back</em>.</p>
        <iframe title="tutorial" width="560" height="315" src="https://www.youtube.com/embed/nOBlJI5w_vI?start=65" frameBorder="0" allow="autoplay; encrypted-media; picture-in-picture" allowFullScreen></iframe>
        <button className='btn green' onClick={ () => props.redirectTo('Game') }>Play</button>
        <button className='btn orange' onClick={ () => props.redirectTo('Menu') }>Menu</button>
    </div>
} 

export default Tutorial
