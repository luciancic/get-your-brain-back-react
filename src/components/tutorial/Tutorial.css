.tutorial {
    height: 100vh;
    display: grid;
    grid-template-rows: 4fr 1fr;
}

.tutorial .game {
    height: 80vh;
}

.roundBar {
    background: linear-gradient(rgba(0, 4, 255, 0), rgba(0, 4, 255, 0.8));
    height: 200px;
    display: grid;
    grid-template-columns: 1fr 1fr 1fr;
    align-items: center;
}

.positionMini {
    position: relative;
    bottom: 75px;
    transform: scale(0.3, 0.3);
}

@media (max-height: 500px) and (min-aspect-ratio: 16/10) {
   
}