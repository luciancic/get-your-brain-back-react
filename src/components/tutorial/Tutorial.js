// Needs to show text above, n, game board, and the match buttons.

// 1st page: "Welcome!: Get your brain back! is a memory game where you need to remember the 
// positions of the circles and the letters you hear. In this tutorial, you will get familiar
// with rounds, n, and matching."

// 2nd page: "Rounds: A round is time during which the circle is visible. When the circle fades
// out, the round ends. When the next circle appears, the next round starts, and so on."

// 3rd page: "N: n is number of rounds you will have to remember at a time.
// The lowest possible n is 2"

// 4th page: "Matching: When the circle appears in the same place as n rounds ago, press the
// match button."

// 5th page: "Advanced matching: Playing with position alone would be too easy; we want a thorough
// memory workout. You will have to watch out for the position and the letter at the same time.
// Be careful though! The position and letters may match at the same time, or not. Remember them
// separately."

// 6th page: "Raising n: As you get better at the game, n will be raised. See what happens with an 
// n of 3."

// 7th page: "Pro tips"

import React, { Component } from 'react';
import { Redirect } from "react-router-dom";
import { connect } from 'react-redux';
import Court from '../court/Court';
import { startTutorial, exitTutorial, nextTutorial, previousTutorial } from '../../actions/tutorialActions';

import "./Tutorial.css";

export class Tutorial extends Component {
    componentDidMount() {
        startTutorial();
    }

    render() {
        return <div className="tutorial">
            <section className="tutorial-info">
                <h1>Intro</h1>
                <p><em>Get your brain back!</em> is a memory game where you need to remember the positions of the circles and the letters you hear. In this tutorial, you will get familiar with rounds, n, and matching.</p>
                <div className="tutorial-buttons">
                    <button onClick={this.props.previousTutorial} className="btn">Back</button>
                    <button onClick={this.props.nextTutorial} className="btn">Next</button>
                    <button onClick={this.props.exitTutorial} className="btn">Quit</button>
                </div>
            </section>
            <Court 
                {...this.props}
                // activeLetter={1} // Placeholder value
                // activePosition={1} // Placeholder value 
                letters={[1]} 
                positions={[1]} 
            />
            { this.props.tutorial.running ? null : <Redirect to="/" /> }
        </div>
    }
} 

function mapStateToProps({ tutorial }) {
    return { tutorial }
}

const actions = {
    startTutorial,
    exitTutorial,
    nextTutorial,
    previousTutorial
}

export default connect(mapStateToProps, actions)(Tutorial)