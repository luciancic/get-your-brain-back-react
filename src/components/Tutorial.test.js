import React from 'react';
import { shallow } from 'enzyme';
import { Tutorial } from './Tutorial';

it('renders Tutorial', () => {
  expect( shallow( <Tutorial tutorial={{}}/> )).toMatchSnapshot();
});