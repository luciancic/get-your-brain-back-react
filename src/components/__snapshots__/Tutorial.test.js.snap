// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Tutorial 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Tutorial
    tutorial={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <section
          className="tutorial-info"
        >
          <h1>
            Intro
          </h1>
          <p>
            <em>
              Get your brain back!
            </em>
             is a memory game where you need to remember the positions of the circles and the letters you hear. In this tutorial, you will get familiar with rounds, n, and matching.
          </p>
          <div
            className="tutorial-buttons"
          >
            <button
              className="btn"
            >
              Back
            </button>
            <button
              className="btn"
            >
              Next
            </button>
            <button
              className="btn"
            >
              Quit
            </button>
          </div>
        </section>,
        <escapeable
          letters={
            Array [
              1,
            ]
          }
          positions={
            Array [
              1,
            ]
          }
          tutorial={Object {}}
        />,
        <Redirect
          push={false}
          to="/"
        />,
      ],
      "className": "tutorial",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              Intro
            </h1>,
            <p>
              <em>
                Get your brain back!
              </em>
               is a memory game where you need to remember the positions of the circles and the letters you hear. In this tutorial, you will get familiar with rounds, n, and matching.
            </p>,
            <div
              className="tutorial-buttons"
            >
              <button
                className="btn"
              >
                Back
              </button>
              <button
                className="btn"
              >
                Next
              </button>
              <button
                className="btn"
              >
                Quit
              </button>
            </div>,
          ],
          "className": "tutorial-info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Intro",
            },
            "ref": null,
            "rendered": "Intro",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <em>
                  Get your brain back!
                </em>,
                " is a memory game where you need to remember the positions of the circles and the letters you hear. In this tutorial, you will get familiar with rounds, n, and matching.",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Get your brain back!",
                },
                "ref": null,
                "rendered": "Get your brain back!",
                "type": "em",
              },
              " is a memory game where you need to remember the positions of the circles and the letters you hear. In this tutorial, you will get familiar with rounds, n, and matching.",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <button
                  className="btn"
                >
                  Back
                </button>,
                <button
                  className="btn"
                >
                  Next
                </button>,
                <button
                  className="btn"
                >
                  Quit
                </button>,
              ],
              "className": "tutorial-buttons",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Back",
                  "className": "btn",
                  "onClick": undefined,
                },
                "ref": null,
                "rendered": "Back",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Next",
                  "className": "btn",
                  "onClick": undefined,
                },
                "ref": null,
                "rendered": "Next",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Quit",
                  "className": "btn",
                  "onClick": undefined,
                },
                "ref": null,
                "rendered": "Quit",
                "type": "button",
              },
            ],
            "type": "div",
          },
        ],
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "letters": Array [
            1,
          ],
          "positions": Array [
            1,
          ],
          "tutorial": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "push": false,
          "to": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <section
            className="tutorial-info"
          >
            <h1>
              Intro
            </h1>
            <p>
              <em>
                Get your brain back!
              </em>
               is a memory game where you need to remember the positions of the circles and the letters you hear. In this tutorial, you will get familiar with rounds, n, and matching.
            </p>
            <div
              className="tutorial-buttons"
            >
              <button
                className="btn"
              >
                Back
              </button>
              <button
                className="btn"
              >
                Next
              </button>
              <button
                className="btn"
              >
                Quit
              </button>
            </div>
          </section>,
          <escapeable
            letters={
              Array [
                1,
              ]
            }
            positions={
              Array [
                1,
              ]
            }
            tutorial={Object {}}
          />,
          <Redirect
            push={false}
            to="/"
          />,
        ],
        "className": "tutorial",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                Intro
              </h1>,
              <p>
                <em>
                  Get your brain back!
                </em>
                 is a memory game where you need to remember the positions of the circles and the letters you hear. In this tutorial, you will get familiar with rounds, n, and matching.
              </p>,
              <div
                className="tutorial-buttons"
              >
                <button
                  className="btn"
                >
                  Back
                </button>
                <button
                  className="btn"
                >
                  Next
                </button>
                <button
                  className="btn"
                >
                  Quit
                </button>
              </div>,
            ],
            "className": "tutorial-info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Intro",
              },
              "ref": null,
              "rendered": "Intro",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <em>
                    Get your brain back!
                  </em>,
                  " is a memory game where you need to remember the positions of the circles and the letters you hear. In this tutorial, you will get familiar with rounds, n, and matching.",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Get your brain back!",
                  },
                  "ref": null,
                  "rendered": "Get your brain back!",
                  "type": "em",
                },
                " is a memory game where you need to remember the positions of the circles and the letters you hear. In this tutorial, you will get familiar with rounds, n, and matching.",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <button
                    className="btn"
                  >
                    Back
                  </button>,
                  <button
                    className="btn"
                  >
                    Next
                  </button>,
                  <button
                    className="btn"
                  >
                    Quit
                  </button>,
                ],
                "className": "tutorial-buttons",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Back",
                    "className": "btn",
                    "onClick": undefined,
                  },
                  "ref": null,
                  "rendered": "Back",
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Next",
                    "className": "btn",
                    "onClick": undefined,
                  },
                  "ref": null,
                  "rendered": "Next",
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Quit",
                    "className": "btn",
                    "onClick": undefined,
                  },
                  "ref": null,
                  "rendered": "Quit",
                  "type": "button",
                },
              ],
              "type": "div",
            },
          ],
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "letters": Array [
              1,
            ],
            "positions": Array [
              1,
            ],
            "tutorial": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "push": false,
            "to": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
